### PR TITLE
8293850: need a largest_committed  metric for each category of NMT's output

### PR DESCRIPTION
**Pull request by afshin-zafari**
The `peak` amount of allocation is kept for Virtual Memory allocations.
Whenever `reserved` and `committed` amounts are printed, the `largest_committed` also is printed.

tiers 1-7 passed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293850](https://bugs.openjdk.org/browse/JDK-8293850): need a largest_committed  metric for each category of NMT's output (**Enhancement** - P4)

### Reviewers
 * [Gerard Ziemski](https://openjdk.org/census#gziemski) (@gerard-ziemski - Committer) ⚠️ Review applies to [1fb0a284](https://git.openjdk.org/jdk/pull/15305/files/1fb0a28404605f42e3e6e63d27010d6a3deefb88)
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - **Reviewer**) ⚠️ Review applies to [1fb0a284](https://git.openjdk.org/jdk/pull/15305/files/1fb0a28404605f42e3e6e63d27010d6a3deefb88)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15305/head:pull/15305` \
`$ git checkout pull/15305`

Update a local copy of the PR: \
`$ git checkout pull/15305` \
`$ git pull https://git.openjdk.org/jdk.git pull/15305/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15305`

View PR using the GUI difftool: \
`$ git pr show -t 15305`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15305.diff">https://git.openjdk.org/jdk/pull/15305.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15305#issuecomment-1687776354)

### CODE REVIEW

**Review by gerard-ziemski — APPROVED** [review]
LGTM thank you for fixing this!

File: src/hotspot/share/services/memReporter.cpp [hunk 1]
```diff
@@ -50,10 +50,13 @@ size_t MemReporterBase::committed_total(const MallocMemory* malloc, const Virtua
   return malloc->malloc_size() + malloc->arena_size() + vm->committed();
 }
 
-void MemReporterBase::print_total(size_t reserved, size_t committed) const {
+void MemReporterBase::print_total(size_t reserved, size_t committed, size_t peak) const {
   const char* scale = current_scale();
   output()->print("reserved=" SIZE_FORMAT "%s, committed=" SIZE_FORMAT "%s",
     amount_in_current_scale(reserved), scale, amount_in_current_scale(committed), scale);
+  if (peak != 0) {
+    output()->print(", largets_committed=" SIZE_FORMAT "%s", amount_in_current_scale(peak), scale);
```
Comment (jdksjolen) [inline src/hotspot/share/services/memReporter.cpp:58]: `largets` => `largest`

**Review by jdksjolen — APPROVED** [review]
LGTM, except there's a typo in `MemReportBase::print_total`'s output.

**Comment by afshin-zafari** [comment]
Thank you Johan and Gerard for your review.
/integrate

**Pull request closed without merging**